File: Mar2026/16thMarch2026/secondLargest.js
function secondLargest(arr){
    let firstLr = -Infinity
    let secondLr = - Infinity

    for(const ele of arr){
        if(ele > firstLr){
            secondLr = firstLr
            firstLr = ele
        }
        else if(ele > secondLr && ele < firstLr){
            secondLr = ele
        }
    }
    return secondLr === -Infinity ? null : secondLr
}

console.log("Second Largest", secondLargest([10, 5, 8, 20, 15]))

File: Mar2026/18thMarch2026/secondLargest.py
def second_larhest(arr):
    first_lr = 0
    second_lr = 0
    for i in arr:
        if(i>first_lr):
            second_lr = first_lr
            first_lr = i
        elif(i>second_lr and i<first_lr):
            second_lr = i
    return second_lr

print(second_larhest([10, 5, 8, 20, 15]))